includes/Admin/Settings.php

```diff
@@ -104,54 +92,58 @@ public function add_menu_item() {
 			[ $this, 'render' ],
 			5
 		);
-		$this->connect_to_enhanced_admin( $is_marketing_enabled ? 'marketing_page_wc-facebook' : 'woocommerce_page_wc-facebook' );
-
-		if ( $is_marketing_enabled ) {
-			$this->add_fb_product_sets_to_marketing_menu();
-		}
+		$this->connect_to_enhanced_admin( $this->is_marketing_enabled() ? 'marketing_page_wc-facebook' : 'woocommerce_page_wc-facebook' );
 	}
 
 	/**
-	 * Checks for connection and if established adds Facebook Product Sets taxonomy page to the Marketing menu.
+	 * Set the parent and submenu file while accessing Facebook Product Sets in the marketing menu.
 	 *
 	 * @since 2.6.29
+	 * @param string $parent_file The parent file.
+	 * @return string
 	 */
-	private function add_fb_product_sets_to_marketing_menu() {
-		$is_connected = facebook_for_woocommerce()->get_connection_handler()->is_connected();
+	public function set_parent_and_submenu_file( $parent_file ) {
+		global $pagenow, $submenu_file, $current_screen;
 
-		// If a connection is not established, do not add Facebook Product Sets to Marketing menu.
-		if ( ! $is_connected ) {
-			return;
+		$root_menu_item = $this->root_menu_item();
+
+		if ( 'edit-tags.php' == $pagenow ) {
```

Comment:
To also show the right menu when editing a product set
```suggestion
		if ( in_array( $pagenow, array( 'edit-tags.php', 'term.php' ), true ) ) {
```